index.js

```diff
@@ -2,12 +2,28 @@
 var vm = require('vm');
 var fs = require('fs');
 var _ = require('lodash');
-var Promise = require('bluebird');
 var path = require('path');
 var childCompiler = require('./lib/compiler.js');
 var prettyError = require('./lib/errors.js');
 var chunkSorter = require('./lib/chunksorter.js');
-Promise.promisifyAll(fs);
+
+function statAsync (filename) {
```

Comment:
Should we use: https://github.com/normalize/mz ?

Reply:
I considered a couple of ways of doing it, the most straightforward I found was the native `util.promisify` (http://2ality.com/2017/05/util-promisify.html) and it's polyfill. 

I settled for just using basic promise functionality because it didn't introduce other dependencies and it seemed the most explicit way of doing it (at the cost of a bit of repetition).

Reply:
> I settled for just using basic promise functionality because it didn't introduce other dependencies and it seemed the most explicit way of doing it (at the cost of a bit of repetition).

yep, I kinds of agree with you :) 
We just need to make a choice: more code or more dependencies
I will let Jan answer that

Reply:
Imho we should rather use a promisify util or mz as they provide already unit tested code and we don't have to maintain that part of code in our project. 
It looks like `mz` is not using native promises: https://github.com/normalize/mz#promise-engine

Reply:
Added `util.promisify` and used it for `statAsync` and `readFileAsync`, but did not use it for `applyPluginsAsyncWaterfall` since it requires being called as a method of `compilation`.